pkg/lib/aws/cloudformation.go

```diff
@@ -23,16 +23,22 @@ import (
 )
 
 func (c *Client) ListEKSStacks(controlPlaneStackName string, nodegroupStackNames strset.Set) ([]*cloudformation.StackSummary, error) {
-	var stackSummaries []*cloudformation.StackSummary
+	stackRecordsByName := map[string][]*cloudformation.StackSummary{}
 	stackSet := strset.Union(nodegroupStackNames, strset.New(controlPlaneStackName))
 
 	err := c.CloudFormation().ListStacksPages(
 		&cloudformation.ListStacksInput{},
 		func(listStackOutput *cloudformation.ListStacksOutput, lastPage bool) bool {
 			for _, stackSummary := range listStackOutput.StackSummaries {
-
-				if stackSet.HasWithPrefix(*stackSummary.StackName) {
-					stackSummaries = append(stackSummaries, stackSummary)
+				if stackSummary == nil || stackSummary.StackName == nil || !stackSet.HasWithPrefix(*stackSummary.StackName) {
+					continue
+				}
+				if _, ok := stackRecordsByName[*stackSummary.StackName]; !ok {
+					stackRecordsByName[*stackSummary.StackName] = []*cloudformation.StackSummary{
+						stackSummary,
+					}
+				} else {
+					stackRecordsByName[*stackSummary.StackName] = append(stackRecordsByName[*stackSummary.StackName], stackSummary)
```

Comment:
Rather than storing all of the old versions, this function can only keep the latest.